plugins/dbqueuedb/dbqueuedb.c

```diff
@@ -501,6 +501,25 @@ static void admin(struct dbenv *dbenv, int type)
     Pthread_mutex_lock(&dbqueuedb_admin_lk);
     dbqueuedb_admin_running = 0;
     Pthread_mutex_unlock(&dbqueuedb_admin_lk);
+} 
+
+static int stat_odh_callback(int consumern, size_t length, unsigned int epoch,
+                                 unsigned int depth, void *userptr)
+{
+    struct consumer_stat *stats = userptr;
+
+    if (consumern < 0 || consumern >= MAXCONSUMERS) {
+        logmsg(LOGMSG_USER, "%s: consumern=%d length=%u epoch=%u\n",
+                __func__, consumern, (unsigned)length, epoch);
+    } else {
+        assert(!stats[consumern].has_stuff);
+        stats[consumern].has_stuff = 1;
+        stats[consumern].first_item_length = length;
```

Comment:
This needs `if (!stats[consumern].has_stuff)` - it's meant to latch the first value and not override it again.  Then again, we don't need to walk anymore with this code, we can just pull out the first and last sequence numbers in get_stats.

Reply:
No- it doesn't walk anymore- it pulls out the first and the last values, filling the original structure with the information.  See how this is called in stat_thread_int->dbq_odh_stats->bdb_queuedb_stats (line 397)

Reply:
Agreed - misread the code.